src/dxgi/dxgi_swapchain.cpp

```diff
@@ -262,6 +264,16 @@ namespace dxvk {
     if (SyncInterval > 4)
       return DXGI_ERROR_INVALID_CALL;
 
+    if (GetWindowLongW(m_window, GWL_STYLE) & WS_MINIMIZE)
+      return DXGI_STATUS_OCCLUDED;
+
+    if (!m_descFs.Windowed && GetForegroundWindow() != m_window)
+    {
+      if (!(PresentFlags & DXGI_PRESENT_TEST))
+        SetFullscreenState(FALSE, nullptr);
+      return DXGI_STATUS_OCCLUDED;
```

Comment:
Is it intended to return `DXGI_STATUS_OCCLUDED` for only one single frame here?

Also, if the window becomes current again, does native DXGI automatically go into fullscreen mode again or does that require app intervention?

Reply:
DXGI_STATUS_OCCLUDED is returned whenever swapchain stays fullscreen while we are out of focus. Yes, in practice that will mean that once we presented and got out of fullscreen the next frame will be presented ok and won't return DXGI_STATUS_OCCLUDED (unless that was DXGI_PRESENT_TEST).  I think this DXGI_STATUS_OCCLUDED return is covered by the linked test. Not sure what is the concern here, or am I probably missing something?

Regarding going fullscreen mode again, I think dxgi doesn't go fullscreen automatically. Also, I don't see a straightforward mechanism in its logic how would it do it. It gets kicked out of fullscreen by loosing exclusive ownership on the output (which is managed by D3DKMT upon detecting that the window got occluded). I can't immediately think of the mechanism which would do the opposite and somehow suggest dxgi that it should go back fullscreen.